autoload/sy/repo.vim

```diff
@@ -342,20 +413,32 @@ endfunction
 " Function: s:initialize_job {{{1
 function! s:initialize_job(vcs) abort
   let vcs_cmd = s:expand_cmd(a:vcs, g:signify_vcs_cmds)
+  return s:wrap_cmd(a:vcs, vcs_cmd)
+endfunction
+
+" Function: s:initialize_buffer_job {{{1
+function! s:initialize_buffer_job(vcs, live) abort
+  let base_cmd = s:get_base_cmd(a:vcs)
+  let diff_cmd = 'set -o pipefail; ' . base_cmd . ' | ' . s:difftool . ' -U0 - ' . fnameescape(a:live)
```

Comment:
From a first glance, `set -o pipefail` probably won't work in shells like fish, Powershell, or cmd.exe.

Reply:
I misunderstood thinking that it was always wrapped in `sh -c` but I now see that is wrong. I don't think there is a portable way to do this however I can just implement using a temp file. The performance difference should be negligible.